pallets/attesters/src/lib.rs

```diff
@@ -0,0 +1,864 @@
+        ) -> DispatchResult {
+            let submitter = ensure_signed(origin)?;
+
+            Batches::<T>::try_mutate(target, |batches_option| {
+                if let Some(batches) = batches_option {
+                    let index = batch_index as usize;
+                    if let Some(batch) = batches.get_mut(index) {
+                        match batch.status {
+                            BatchStatus::Pending | BatchStatus::ReadyForSubmission => {
+                                batch.status = BatchStatus::Committed;
+                                Ok(Some(batches.clone()))
+                            },
+                            _ => Err(Error::<T>::BatchAlreadyCommitted),
+                        }
+                    } else {
+                        Err(Error::<T>::BatchNotFound)
+                    }
+                } else {
+                    Err(Error::<T>::BatchNotFound)
+                }
+            })?;
+
+            // Reward the submitter
+            let fast_confirmation_cost =
+                FastConfirmationCost::<T>::get(target).unwrap_or_else(Zero::zero);
+            let total_reward = fast_confirmation_cost.saturating_mul(T::RewardMultiplier::get());
+
+            T::Currency::transfer(
+                &T::CommitmentRewardSource::get(),
+                &submitter,
+                total_reward,
+                ExistenceRequirement::KeepAlive,
+            )?;
+
+            Ok(())
+        }
+
+        #[pallet::weight(10_000)]
+        pub fn nominate(
+            origin: OriginFor<T>,
+            attester: T::AccountId,
+            amount: BalanceOf<T>,
+        ) -> DispatchResult {
+            let nominator = ensure_signed(origin)?;
+            Self::do_nominate(nominator.clone(), attester.clone(), amount)?;
+            Self::deposit_event(Event::Nominated(nominator, attester, amount));
+            Ok(())
+        }
+    }
+
+    impl<T: Config> Pallet<T> {
+        pub fn committee_size() -> usize {
+            T::CommitteeSize::get() as usize
+        }
+
+        pub fn do_nominate(
+            nominator: T::AccountId,
+            attester: T::AccountId,
+            amount: BalanceOf<T>,
+        ) -> DispatchResult {
+            let mut nomination = match Nominations::<T>::get(&attester) {
+                Some(nomination) => nomination,
+                None => (attester.clone(), Zero::zero()),
+            };
+
+            // Check if nominator has enough balance
+            ensure!(
+                T::Currency::free_balance(&nominator) >= amount,
+                Error::<T>::NominatorNotEnoughBalance
+            );
+
+            nomination.1 += amount;
+
+            let nomination_amount = nomination.1;
+
+            // Update the nomination storage item
+            Nominations::<T>::insert(&attester, nomination);
+
+            // Update the sorted list of nominated attesters
+            SortedNominatedAttesters::<T>::try_mutate(|attesters| {
```

Comment:
I think we should only trigger after a certain period is complete, in this form there are some major slashing implications